src/dune_rules/pkg_rules.ml

```diff
@@ -1062,9 +1352,11 @@ module Action_expander = struct
     ;;
   end
 
-  let expander context (pkg : Pkg.t) =
+  let expander context (pkg : Pkg.t) ~override_pform =
     let+ { Artifacts_and_deps.binaries; dep_info } =
-      Pkg.deps_closure pkg |> Artifacts_and_deps.of_closure
+      Pkg.deps_closure pkg
+      |> List.filter ~f:(Fun.negate Pkg.is_system_provided)
```

Comment:
how come this filter is needed now?

Reply:
Otherwise dune would attempt to install system provided packages, which we don't want it to do.